File: tests/0125_valid_palindrome_test.py
import importlib
import pytest

module = importlib.import_module("algorithms.0125_valid_palindrome")


def test_valid_palindrome():
    s = module.Solution()
    assert s.isPalindrome("A man, a plan, a canal: Panama")
    assert s.isPalindrome("")
    assert s.isPalindrome("s")
    assert s.isPalindrome("ss")
    assert s.isPalindrome("sAs")
    assert not s.isPalindrome("race a car")

File: tests/0125_valid_palindrome_test.cpp
#include "0125_valid_palindrome.cpp"
#include <doctest.h>

TEST_CASE("ValidParenthesesTests")
{
    Solution s = Solution();
    CHECK(s.isPalindrome("A man, a plan, a canal: Panama"));
    CHECK(s.isPalindrome(""));
    CHECK(s.isPalindrome("s"));
    CHECK(s.isPalindrome("ss"));
    CHECK(s.isPalindrome("sAs"));
    CHECK_FALSE(s.isPalindrome("race a car"));
}
